hadoop-hdds/docs/content/design/dn-min-space-configuration.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Minimum free space configuration for datanode volumes
+summary: Describe proposal for minimum free space configuration which volume must have to function correctly.
+date: 2025-05-5
+jira: HDDS-12928
+status: draft
+author: Sumit Agrawal
+---
+<!--
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+
+# Abstract
+Volume in datanode stores the container data and metadata of this is present in volume db.
+There are various parallel operation going on such as import container, export container, write and delete data blocks,
```

Comment:
Also, rebuilds of EC containers might also be on going. I like @krishnaasawa1's suggestion of bringing
in the container size into the calculation. Maybe `max(min disk space, configured spare container multiple*size per container)`. The default container multiple being 4 (20 gigs)

Reply:
Above is good opinion, but just considering below with usecase, it will be handled with committed space logic.

When any container operation is going on, it must reserve committed space (ie space is may consume once container is build). With this, need for extra container space may be not be required. 

Will re-check EC container rebuild for committed space or not to fix specific part, And any possibility of crossing min-avaialable space.

Reply:
Verified, we do increase committed space when create container in recovering state for EC.